101-rds-deployment-full-ha/azuredeploy.json

```diff
@@ -0,0 +1,1180 @@
+                    "projectName": {
+                        "value": "[parameters('projectName')]"
+                    },
+                    "location": {
+                        "value": "[parameters('location')]"
+                    },
+                    "TimeZoneID": {
+                        "value": "[parameters('TimeZoneID')]"
+                    },
+                    "loopCount": {
+                        "value": "[parameters('vmProperties')[copyIndex()].count]"
+                    },
+                    "storageDiagUrl": {
+                        "value": "[reference(concat(parameters('projectName'),'stgdiag')).primaryEndpoints.blob]"
+                    },
+                    "vmName": {
+                        "value": "[parameters('vmProperties')[copyIndex()].name]"
+                    },
+                    "subNetRef": {
+                        "value": "[variables('subNetRef')]"
+                    },
+                    "vmSize": {
+                        "value": "[parameters('vmSize')]"
+                    },
+                    "storageSkuType": {
+                        "value": "[parameters('storageSkuType')]"
+                    },
+                    "adminUser": {
+                        "value": "[parameters('adminUser')]"
+                    },
+                    "adminPasswd": {
+                        "value": "[parameters('adminPasswd')]"
+                    },
+                    "intLbName": {
+                        "value": "[concat(parameters('projectName'),'intlb')]"
+                    },
+                    "intLbBackEndPool": {
+                        "value": "[parameters('vmProperties')[copyIndex()].intLbBackEndPool]"
+                    },
+                    "intLbBrokerIP": {
+                        "value": "[reference(resourceId('Microsoft.Network/loadBalancers',concat(parameters('projectName'),'intlb'))).frontendIPConfigurations[0].properties.privateIPAddress]"
+                    },
+                    "intLbWebGWIP": {
+                        "value": "[reference(resourceId('Microsoft.Network/loadBalancers',concat(parameters('projectName'),'intlb'))).frontendIPConfigurations[1].properties.privateIPAddress]"
+                    },                    
+                    "pubLbName": {
+                        "value": "[concat(parameters('projectName'),'publb')]"
+                    },
+                    "pubLbBackEndPool": {
+                        "value": "[parameters('vmProperties')[copyIndex()].pubLbBackEndPool]"
+                    },
+                    "adDomainName": {
+                        "value": "[parameters('adDomainName')]"
+                    },
+                    "firstDcIP": {
+                        "value": "[variables('firstDcIP')]"
+                    },
+                    "MainConnectionBroker": {
+                        "value": "[concat(parameters('vmProperties')[2].name,'1')]"

```

Comment:
this pattern will not work if I change the parameter values to have more/fewer or a different order of vm properties...

Given the number of issue with the pattern, I think we should move the array params in the parent template to variables, so that we can reduce chances for failure due to the values users enter.

Reply:
still needs to be addressed